src/plugins/chart_expressions/expression_metric/public/components/metric_vis.tsx

```diff
@@ -0,0 +1,236 @@
+};
+
+const getColor = (value: number, paletteParams: CustomPaletteState | undefined) =>
+  paletteParams
+    ? getPaletteService().get('custom')?.getColorForValue?.(value, paletteParams, {
+        min: paletteParams.rangeMin,
+        max: paletteParams.rangeMax,
+      }) || defaultColor
+    : defaultColor;
+
+export interface MetricVisComponentProps {
+  data: Datatable;
+  config: Pick<VisParams, 'metric' | 'dimensions'>;
+  renderComplete: IInterpreterRenderHandlers['done'];
+}
+
+const MetricVisComponent = ({ data, config, renderComplete }: MetricVisComponentProps) => {
+  const primaryMetricColumn = getColumnByAccessor(config.dimensions.metric, data.columns)!;
+  const formatPrimaryMetric = getFormatter(config.dimensions.metric, data.columns);
+
+  let secondaryMetricColumn;
+  let formatSecondaryMetric: ReturnType<typeof getFormatter>;
+  if (config.dimensions.secondaryMetric) {
+    secondaryMetricColumn = getColumnByAccessor(config.dimensions.secondaryMetric, data.columns);
+    formatSecondaryMetric = getFormatter(config.dimensions.secondaryMetric, data.columns);
+  }
+
+  const breakdownByColumn = config.dimensions.breakdownBy
+    ? getColumnByAccessor(config.dimensions.breakdownBy, data.columns)
+    : undefined;
+
+  const metricConfigs: MetricSpec['data'][number] = [];
+
+  const useProgressBar =
+    typeof config.metric.progressMin === 'number' && typeof config.metric.progressMax === 'number';
+
+  const commonProps = {
+    valueFormatter: formatPrimaryMetric,
+    ...(useProgressBar
+      ? {
+          domain: { min: config.metric.progressMin, max: config.metric.progressMax },
+          progressBarDirection: config.metric.progressDirection,
+        }
+      : {}),
+  };
+
+  if (!breakdownByColumn) {
+    const value = data.rows[0][primaryMetricColumn.id];
+
+    metricConfigs.push({
+      ...commonProps,
+      value,
+      title: primaryMetricColumn.name,
+      subtitle: secondaryMetricColumn?.name ?? config.metric.subtitle,
+      extra: (
+        <span>
+          {secondaryMetricColumn
+            ? formatSecondaryMetric!(data.rows[0][secondaryMetricColumn.id])
+            : config.metric.extraText}
+        </span>
+      ),
+      color: getColor(value, config.metric.palette),
+    });
+  }
+
+  if (breakdownByColumn) {
+    for (const row of data.rows) {
+      const value = row[primaryMetricColumn.id];
+
+      metricConfigs.push({
+        ...commonProps,
+        value,
+        title: row[breakdownByColumn.id],
+        subtitle: primaryMetricColumn.name,
+        extra: (
+          <span>
+            {secondaryMetricColumn
+              ? formatSecondaryMetric!(row[secondaryMetricColumn.id])
+              : config.metric.extraText}
+          </span>
+        ),
+        color: getColor(value, config.metric.palette),
+      });
+    }
+  }
+
+  if (config.metric.minTiles) {
+    while (metricConfigs.length < config.metric.minTiles) metricConfigs.push(undefined);
```

Comment:
@markov00 @gvnmagni Any opinion on how the "empty" cells should be rendered? Right now it's just a blank square which is probably a good start but just to make sure
<img width="878" alt="Screenshot 2022-07-06 at 13 04 03" src="https://user-images.githubusercontent.com/1508364/177536579-086160ec-21ad-4488-8a06-ffd8523644e2.png">

Reply:
That is perfect! It is actually what we expected to see

<img width="780" alt="Screenshot 2022-07-07 at 09 33 12" src="https://user-images.githubusercontent.com/18484576/177717594-febfeb41-f183-4556-b663-eb9797be5187.png">